### PR TITLE
fix: [CDS-48176]: show cluster API error when there is error returned from API

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
### Summary
- Cluster API error will be shown when user clicks on Cluster field
- Other similar changes in runtime view

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by chetanvekariya-harness** [comment]
retrigger coverage

File: src/modules/75-cd/components/PipelineSteps/ECSInfraSpec/ECSInfraSpecEditable.tsx [hunk 1]
```diff
@@ -84,27 +95,72 @@ export const ECSInfraSpecEditable: React.FC<ECSInfraSpecEditableProps> = ({
...
+      if (canFetchClusters(connectorRef, region)) {
+        setLastQueryData({ connectorRef, region })
+      }
+    },
+    [canFetchClusters, lastQueryData]
+  )
 
   const validationSchema = getECSInfraValidationSchema(getString)
 
+  const itemRenderer = React.useCallback(
```
Comment (satyam-harness) [inline src/modules/75-cd/components/PipelineSteps/ECSInfraSpec/ECSInfraSpecEditable.tsx:147]: We have a component for it now 
`src/modules/10-common/components/ItemRenderer/ItemRendererWithMenuItem.tsx`

File: src/modules/75-cd/components/PipelineSteps/ECSInfraSpec/ECSInfraSpecInputForm.tsx [hunk 2]
```diff
@@ -82,31 +85,79 @@ const ECSInfraSpecInputForm = ({
...
+            envId: environmentRef,
+            infraDefinitionId: infrastructureRef
+          }
+        })
+      }
+    },
+    [canFetchClusters, lastQueryData]
+  )
+
+  const itemRenderer = React.useCallback(
```
Comment (satyam-harness) [inline src/modules/75-cd/components/PipelineSteps/ECSInfraSpec/ECSInfraSpecInputForm.tsx:145]: same here
`src/modules/10-common/components/ItemRenderer/ItemRendererWithMenuItem.tsx`

File: src/modules/75-cd/components/PipelineSteps/ECSServiceSpec/ManifestSource/S3ManifestStoreRuntimeView.tsx [hunk 3]
```diff
@@ -185,6 +189,23 @@ export const S3ManifestStoreRuntimeView = (props: S3ManifestStoreRuntimeViewProp
     )
   }
 
+  const itemRenderer = React.useCallback(
```
Comment (satyam-harness) [inline src/modules/75-cd/components/PipelineSteps/ECSServiceSpec/ManifestSource/S3ManifestStoreRuntimeView.tsx:192]: same here

**Review by satyam-harness — APPROVED** [review]

**Review by litvinovstas — APPROVED** [review]

**Pull request merged**